### ISSUE TITLE
Consume package in node_modules using ESM modules

### ISSUE THREAD

**Issue by jonbri** [slinkity/slinkity#221]
## Describe the bug
I'm trying to use a third-party component library that is delivered as ES modules.

## To Reproduce
1. Start a new react-flavored slinkity project ([quick start](https://slinkity.dev/docs/quick-start/))
2. Install package: `yarn add tick-tick`
3. Import the esm modules: `import { TickTick } from "tick-tick/esm/index";`
4. Run `npm run dev`
5. See error

## Expected behavior
I should be able to `import` modules from the package.

 - OS: [Windows]
 - Browser [Chrome]
 - Version [0.8.2]

**Additional context**
Here is the error I get initially:
```
[Browsersync] Serving files from: _site
@my-esm/package doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point). Please contact the package author to fix.
2:07:57 PM [vite] Error when evaluating SSR module /@fs/home/jonbri/my-slinkity-site/node_modules/@babel/runtime/helpers/extends.js?v=2589198d:
ReferenceError: module is not defined
```

If I go ahead and `"type": "module"` to the third-party package's `package.json` I get this error message:
```
[Browsersync] Serving files from: _site
2:29:16 PM [vite] Error when evaluating SSR module /home/me/my-slinkity-site/src/_includes/Baz.jsx:
Error [ERR_UNSUPPORTED_DIR_IMPORT]: Directory import '/home/me/my-slinkity-site/node_modules/@my-esm/package/components/Button' is not supported resolving ES modules imported from /home/me/my-slinkity-site/node_modules/@my-esm/package/index.js
    at new NodeError (internal/errors.js:322:7)
    at finalizeResolution (internal/modules/esm/resolve.js:314:17)
    at moduleResolve (internal/modules/esm/resolve.js:776:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:887:11)
    at Loader.resolve (internal/modules/esm/loader.js:89:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
    at link (internal/modules/esm/module_job.js:75:36)
```

Is it possible to consume esm modules from `node_modules`?
I assume we can because we can use esm import syntax in our local files (for example, within the `_includes` dir), but I'm nots seeing anything about this in the Slinkity doc.

**Comment by bholmesdev** [comment]
Hey, thanks for reporting this! Agreed this _should_ be supported just fine. To confirm, would you mind replicating in a Stackblitz? [Feel free to fork this one](https://stackblitz.com/edit/vitejs-vite-rujkwj?file=my-slinkity-site%2Fpackage.json&terminal=dev), and run `cd my-slinkity-site && npm start` to start the dev server (sorry, couldn't remove the nested directory here!)

Also curious if this issue persists in a [plain Vite project](https://vitejs.dev/guide/)?

**Comment by jonbri** [comment]
Thanks for the quick reply.
I messed around a little bit with the Stackblitz but didn't see a "my-slinkity-site" directory to cd into.

I published a [sample npm package](https://www.npmjs.com/package/tick-tick) which has both a cjs and esm deliverable to make it easier to rep.
This sample package contains a Button component which renders a number that increments when the button is clicked.
[Here ](https://github.com/jonbri/tick-tick/tree/master/dist) is what is published to NPM:
```
tick-tick
  - cjs
    - CommonJS files
  - esm
    - ESM files
  - index.d.ts
  - package.json
```

In short, consuming the `cjs` folder works, while using the `esm` directory results in the bug.

I put together a gif which demonstrates the problem:
![slinkityRep](https://user-images.githubusercontent.com/1750227/169070635-3526bca9-a2b3-4bdc-ab63-31f73d651224.gif)

- Start out with simple project working
- The "tick-tick" package is consumed as commonjs
- Switch to using the esm module
- Error occurs in console

**Comment by jonbri** [comment]
I also confirmed in a sample Vite React project that both module types work.
The sample components render and are interactive:
```
 import { TickTick as ESMTickTick } from 'tick-tick/esm'
 import { TickTick as CJSTickTick } from 'tick-tick/cjs'
```

**Comment by jonbri** [comment]
I'm now thinking that the issue is related to vite-ssr expecting CJS modules (since it's running in the server).
Once I supply the proper `main` and `module` package.json attributes the issue was resolved.

**Comment by bholmesdev** [comment]
Ah HA good to know `module` resolves the issue @jonbri! I would've suggested [Vite's `ssr.noExternal` config option](https://vitejs.dev/config/#ssr-options) to force CJS bundling. This is a well-known Vite issue that the Astro team's been working to document.
